### PR TITLE
Document CLI batch mode

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Add a new `Batch mode` section to the CLI docs to describe all the remaining options. Also, add the new `--no-progress` variant in the interactive-mode options table.

This is a follow-up to #11894

This is how it looks like:
image

I intentionally made the existing `Output formats` section a subsection of `Batch mode`:
image

> Is this change a fix, improvement, new feature, refactoring, or other?

docs

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

docs

> How would you describe this change to a non-technical end user or system administrator?

n/a

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/installation/cli.rst
Comment (hashhar) [inline docs/src/main/sphinx/installation/cli.rst:392]:
```suggestion
, and prints results in the ``CSV`` format by default.
```

File: docs/src/main/sphinx/installation/cli.rst
Comment (hashhar) [inline docs/src/main/sphinx/installation/cli.rst:147]: `progress` isn't documented already so it needs to be there and it should include mentioning that it can be negated instead of separate entry altogether maybe.

Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst:147]: The options here are slightly modified and broken down into parts, it's not just a copy of the output of `--help`.

File: docs/src/main/sphinx/installation/cli.rst
Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst:392]: There's a comma at the end of the previous line. The whole sentence describes the default behavior in contrast to the interactive mode, so I'm not sure about adding `by default` at the end. It might confuse people about what default we're talking about.

Comment (hashhar) [reply docs/src/main/sphinx/installation/cli.rst:392]: That's a fair point. The table does mention that the output format is configurable.

**Comment by mosabua** [comment]
Hahah .. thats perfect .. we planned to do that soon and now we can just review this ;-)

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:147]: Agree with your setup and comments @nineinchnick

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:148]: query processing progress

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:389]: , or passing

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:421]: non-interactive

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:390]: uses the batch (non-interactive) mode. In this mode the CLI does not report progress, and exists after processing the supplied queries. Results are printed in ``CSV`` format by default. You can configure other formats and redirect the output to a file.

Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:390]: I think we should add an example for that redirection/piping into a file...

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:404]: can you do multiple statements actually?

Also ... should we not reword to process the specified queries ..

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:408]: Do not use ().

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:411]: Specify the format to use for printing query results. 

"for batch mode" is implied and can be left out

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:413]: Clarify that this does NOT affect the output .. at least I assume so .. also how does that work when piping into a file

**Review by mosabua — CHANGES_REQUESTED** [review]
A couple of things need to be clarified imho

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:387]: Change to "Output formats"

File: docs/src/main/sphinx/installation/cli.rst
Comment (hashhar) [reply docs/src/main/sphinx/installation/cli.rst:147]: `--no-progress` doesn't automatically imply that there is a `--progress` flag that can be used. If we are documenting one of them we can mention the other too. Something like `You can use --progress to show query progress` (since that's not default in batch mode AFAIK).

File: docs/src/main/sphinx/installation/cli.rst
Comment (hashhar) [reply docs/src/main/sphinx/installation/cli.rst:390]: That example varies across shells and operating systems combinations. Let's leave it to the shell's documentation instead IMO.

File: docs/src/main/sphinx/installation/cli.rst
Comment (hashhar) [reply docs/src/main/sphinx/installation/cli.rst:404]: Technically we should be calling these statements. Not every SQL statement returns a result and hence is not a query.

Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst:404]: @hashhar is right and I did not reword this, just copied from the output of `--help`. Being able to execute multiple statements is one of the coolest things about the CLI client.

File: docs/src/main/sphinx/installation/cli.rst
Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst:413]: Yeah, looks like it's printed to stderr. Lines updates are somehow messed up when redirecting to a file, no matter if there's a TTY or not (piping vs --execute):
```

jwas@janmac ~/src/trino (git)-[cli-docs-update] % trino --server localhost:8080 --user trino-client --catalog hive --schema v2 --execute 'select count(*) from tpch.sf100.orders;' --progress
"150000000"

Query 20220422_091355_00004_mit6n, FINISHED, 1 node
Splits: 21 total, 21 done (100.00%)
59.08 [150M rows, 0B] [2.54M rows/s, 0B/s]

jwas@janmac ~/src/trino (git)-[cli-docs-update] % trino --server localhost:8080 --user trino-client --catalog hive --schema v2 --execute 'select count(*) from tpch.sf100.orders;' --progress > out.txt
Apr 22, 2022 11:15:02 AM org.jline.utils.Log logr
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
Query 20220422_091502_00005_mit6n [RUNNING] i[0 0B 0B] o[0 0B 0B] splits[0/21/0]
Query 20220422_091502_00005_mit6n, FINISHED, 1 node
Splits: 21 total, 21 done (100.00%)
51.28 [150M rows, 0B] [2.93M rows/s, 0B/s]

jwas@janmac ~/src/trino (git)-[cli-docs-update] % cat out.txt 
"150000000"
jwas@janmac ~/src/trino (git)-[cli-docs-update] % echo 'select count(*) from tpch.sf100.orders;' | trino --server localhost:8080 --user trino-client --catalog hive --schema v2 --progress                              
"150000000"422_091618_00006_mit6n [RUNNING] i[75M 0B 0B] o[75M 0B 0B] splits[0/19/2]

Query 20220422_091618_00006_mit6n, FINISHED, 1 node
Splits: 21 total, 21 done (100.00%)
1:01 [150M rows, 0B] [2.47M rows/s, 0B/s]
```
Important thing is that it does not affect the output.

File: docs/src/main/sphinx/installation/cli.rst
Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst]: Done!

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:416]: ok .. maybe we want to clarify more what happens but in any case.. this is good to go for starters..

File: docs/src/main/sphinx/installation/cli.rst
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst]: Lets add that important fact..

Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst]: I did, see the latest comment. I'll mark this as resolved since it's attached at the old diff.

**Review by mosabua — APPROVED** [review]
Nice work.

**Comment by mosabua** [comment]
Any chance we get this in @hashhar or @electrum ?

File: docs/src/main/sphinx/installation/cli.rst
Comment (hashhar) [inline docs/src/main/sphinx/installation/cli.rst:423]: Link to `Output formats` section?

**Review by hashhar — APPROVED** [review]

File: docs/src/main/sphinx/installation/cli.rst
Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst:423]: I had a link here but @mosabua suggested a change without it. Should I add it back?

Comment (hashhar) [reply docs/src/main/sphinx/installation/cli.rst:423]: IMO it helps future proof since sections may get added or reordered over time and we won't notice then. The link doesn't hurt I think.

Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst]: Done! I'm posting a screenshot to make sure I actually tested the changes :D
image

**Review by hashhar — APPROVED** [review]
Thanks, will merge once CI is done.

**Comment by mosabua** [comment]
So good!

**Pull request merged**
